mhs-reference-implementation/mhs/SDSLookup/tests/test_sdsclient.py

```diff
@@ -0,0 +1,68 @@
+from unittest import TestCase
+import mhs.SDSLookup.sds as sds
+import mhs.SDSLookup.tests.ldap_mocks as mocks
+from utilities.test_utilities import async_test
+
+NHS_SERVICES_BASE = "ou=services, o=nhs"
+
+MHS_OBJECT_CLASS = "nhsMhs"
+
+PARTY_KEY = "A91461-9199084"
+SERVICE_ID = "urn:nhs:names:services:psis:MCCI_IN010000UK13"
+ODS_CODE = "A91461"
+
+
+class TestSDSClient(TestCase):
+
+    @async_test
+    async def test_mhs_details_lookup(self):
+        server, connection = mocks.fake_ldap_server()
+
+        client = sds.SDSClient()
+        client.server = server
+        client.connection = connection
+
+        result = await client._mhs_details_lookup(PARTY_KEY, SERVICE_ID)
+
+        self.assertIsNotNone(result)
+        self.assertEqual(result[0]['attributes']['nhsMHSEndPoint'], ['https://vpn-client-1411.opentest.hscic.gov.uk/'])
+
+    @async_test
+    async def test_accredited(self):
+        server, connection = mocks.fake_ldap_server()
+        client = sds.SDSClient()
+        client.server = server
+        client.connection = connection
+
+        result = await client._accredited_system_lookup(ODS_CODE, SERVICE_ID)
+        assert result
+        assert result[0]['attributes']['nhsMHSPartyKey'] == 'A91461-9199084'
+
+    @async_test
+    async def test_get_mhs_lookup(self):
+        server, connection = mocks.fake_ldap_server()
+        client = sds.SDSClient()
+        client.server = server
+        client.connection = connection
+
+        result = await client.get_mhs_details(ODS_CODE, SERVICE_ID)
+        self.assertIsNotNone(result)
+        self.assertEqual(result['attributes']['nhsMHSPartyKey'], 'A91461-9199084')
```

Comment:
We need to make sure this result has the full set of attributes, as we want to verify it's done the correct two-stage lookup.

There are also some other bits of logic in `get_mhs_details` that we should cover with these tests. It filters results if there is more than one, so we should verify it does that as expected, and it also raises an exception if there are no results(?).

Reply:
Good point, I've updated this to match the other tests - it now checks for all the attributes and verifies no other attributes are included. 

I'm not totally sure we can verify it filters the multiple responses condition, since we can't reliably determine the order or the endpoints. I have added a test for the exception case where there are no responses though.

Reply:
I think the fact that our dummy LDAP response contains multiple results for the same ODS code/party key is enough to cover the filtering.